libbeat/common/docker/watcher.go

```diff
@@ -132,6 +132,12 @@ func NewWatcher(host string, tls *TLSConfig, storeShortID bool) (Watcher, error)
 		return nil, err
 	}
 
+	// extra check to confirm that Docker is available
+	_, err = client.Info(context.Background())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I wonder if this can affect other users of `NewWatcher`, it seems safe though.

Reply:
Yes it could affect others (`AutodiscoverBuilder`) in case of a connectivity error, but it wont be a problem I think, since an error will be returned for a case that is indeed problematic.